[www] Refactor Homepage and Navigation & convert diagram to html/css

fk opened the pull request:
WiP still unfortunately, but getting close! Little time right now, but will follow up with a decent list of what's changed and why later on … let's get this preview built now :P

Blockers:

* Masthead background still has issues on iPad/iPad Pro in portrait mode
* Diagram font sizes for large screens
* Homepage blog posts need a little more love

KyleAMathews commented:
Deploy preview failed.

Built with commit 8e85004d83a8a8b7d1538bbd9d29319dc4772650

https://app.netlify.com/sites/using-remark/deploys/59778d38cf321c76f2bbdc6d

KyleAMathews commented:
Deploy preview failed.

Built with commit 8e85004d83a8a8b7d1538bbd9d29319dc4772650

https://app.netlify.com/sites/gatsbyjs/deploys/59778d36cf321c76f2bbdc4f

KyleAMathews commented:
Deploy preview failed.

Built with commit 1ab61f735e28a077a5fe5747246df890096e0549

https://app.netlify.com/sites/image-processing/deploys/5977906f0752d05a025f6961

KyleAMathews commented:
Deploy preview failed.

Built with commit 1ab61f735e28a077a5fe5747246df890096e0549

https://app.netlify.com/sites/using-styled-components/deploys/5977906f0752d05a025f6967

KyleAMathews commented:
Deploy preview failed.

Built with commit 1ab61f735e28a077a5fe5747246df890096e0549

https://app.netlify.com/sites/using-remark/deploys/5977906f0752d05a025f696c

KyleAMathews commented:
Deploy preview failed.

Built with commit 1ab61f735e28a077a5fe5747246df890096e0549

https://app.netlify.com/sites/using-contentful/deploys/5977906f0752d05a025f6969

KyleAMathews commented:
Deploy preview failed.

Built with commit d1f7f810f5ed37828d662b9604a5f14f7729a06e

https://app.netlify.com/sites/using-postcss-sass/deploys/5977af11424ef2274f48eac7

KyleAMathews commented:
All these Netlify failures seem like their internal problems.

This build for GatsbyJS succeeded however https://5977af10424ef2274f48eabb--gatsbyjs.netlify.com/

Looking *really* cool.

fk commented:
I fixed a few more things, but won't be able to do much more tonight than at least to explain what's all the noise in here:

I originally started off with the intention to fix the whitespace around the homepage navigation – along came an icon replacing the "Github" link and links to the Discord Reactiflux Gatsby channel and the GatsbyJS Twitter:

![image](https://user-images.githubusercontent.com/21834/28598774-345d005e-71a6-11e7-86ce-1411e574bd32.png)

Then I somehow ended up redoing the homepage masthead background as inline SVGs (was linear-gradient before) … this is not super precise either, but for me it lowers the "magic gradient stop number" issues of the previous approach a little.  
Small bonus: An implementation of the layout by @SachaG where the brightest of the three shapes stays attached to the "chopped edge" in the Gatsby logo (as discussed [here](https://github.com/gatsbyjs/gatsby/pull/1355#issuecomment-312741791)).

Then I went to procrastination wonderland:

* Nudged the existing six blocks of copy around – tried [cards](https://cdn.discordapp.com/attachments/313693200269312003/339108267273879553/image.png), but decided for a simpler layout instead that simply tries to do well for the existing copy.
* Adjusted the first two blocks of copy, "Modern web tech…" and "Bring your own data" a little to try and make things a little shorter – without changing the content significantly. Please check @KyleAMathews if this is OK for you.
* Stole the idea and some of the [Reactiflux](https://www.reactiflux.com/) icons, added a Webpack one, and "Mustard" as accent color as suggested by @ArchieHicklin [earlier](https://github.com/gatsbyjs/gatsby/issues/1173#issuecomment-309415650). Right now they don't do much, but I figured they would be useful somehow along the way, plus I love those sprinkles of mustard, so I kept them in. Little did I know they would come in handy again just a day later already.
* Slowly started refactoring styles and moving stuff to components.
* Added the new brand colors to `utils/presets`, and replaced hopefully all inline hex values (apart from those in SVGs not inlined).
* Also added a border radius to the CTA buttons and the main content container on the homepage – these can also be switched of via a value in `presets`.
* Wondered how long it would take to redo the [Explanatory Diagram](https://github.com/gatsbyjs/gatsby/issues/1172) in CSS. Tried a top-to-bottom only version that went quite fast – it's not 100% true to the original though, and could be way nicer. But it does work quite well on small screens, and adjusting to the original left-to-right layout of boxes for bigger screens should be relatively easy. (That said, …we could just have scrolled the image overflow on smaller screens. ;-)) A big nod and bow down to [Segment](https://segment.com/) and their beautiful "Capture data from every customer touch point." diagram design. I got a little jealous of their beautiful SVG animation, so now we have that whimsical 5-minute CSS animation there in the background of the "Build" segment of the Gatsby diagram. 45deg tho! ;-)

![image](https://user-images.githubusercontent.com/21834/28599767-7ec8563c-71ad-11e7-9d5a-e38dce94bc96.png)

One thing that also happened along the way is that I removed the bold underlines from the sidebar links – that didn't happen very conciously though, so now that I look at all that purple …

![image](https://user-images.githubusercontent.com/21834/28599826-f857319e-71ad-11e7-8748-6287eb4eeb6c.png)

That said, I'm also not too sure about those rounded edges any more. :-D

KyleAMathews commented:
This is sooo cool! It's beginning to look really really polished.

A few notes.

* The tech icons sit a bit low on the line
* Looking at segments diagram, their little animated lines are really cool and would work too here. Put a bit more space between the data and Gatsby sections and animate it. Don't want to hold up this PR as everything here is great but this could be a nice refinement.
* There's some placeholder content still at the top of the diagram.

KyleAMathews commented:
Looking at latest build (https://597908907960b1275563eed4--gatsbyjs.netlify.com/) at the diagram and was thinking it'd be nice to change "YAML/JSON" to "YAML, JSON, CSV". CSV is a hugely important way of storing data and I think this will be an attractive feature for a lot of people.

KyleAMathews commented:
Actually, thinking some more.

Make the center box "CMSs" and the last header "Data" and the text inside the box "APIs, Databases, YAML, JSON, CSV, and more!"

Also make the order CMSs, Markdown, Data.

fk commented on www/src/components/gutters.js:
```diff
@@ -0,0 +1,7 @@
+import presets from "../utils/presets"
+import { rhythm } from "../utils/typography"
+
+export const vP = rhythm(presets.gutters.default)
```
@KyleAMathews gutters.js only exist because I ran into an error when trying to import rhythm right in `utils/preset` – there it got me a "rhythm is not a function". I believe I found a similar issue  … yes, here it is: https://github.com/gatsbyjs/gatsby-starter-default/issues/27 … Gotta fix those silly constant names btw.

KyleAMathews commented:
@fk said in chat this is ready to go! Merging away! Thanks for the great work Florian! Looks really nice.

KyleAMathews commented:
@DSchau oh hmmm looks like the image sharp tests are failing as you added a test relying on an image in www which this PR removed. Could you just add an image directly to the __tests__ directory for use there?

fk commented:
@KyleAMathews Ah damn … the Argyle pattern!? ;-)
@DSchau Sorry!

SachaG commented:
Wow I love the new site. It's pretty cool to see how my original 5-min mockup is turning into a proper, well-thought-out design under your hands! I wish every design project was this easy ;)

SachaG commented:
Oh and I think my favorite detail is the little emoji-like icons for each project. I need to steal that :)

fk commented:
@SachaG Wow, thank you Sacha for your nice words, that's a huuuge compliment! Very glad to have your approval, that means a lot! That said, now let me fix those remaining breakpoint issues with the icons top right 🙄

Btw., while doing this is discussed dropping one of the stripes in the right part of the design for small screens to have a little better control over the bright-on-dark contents once they're there with @KyleAMathews:

![image](https://user-images.githubusercontent.com/21834/28670960-48d6207a-72db-11e7-9dc4-cbfa50a6094d.png)

![image](https://user-images.githubusercontent.com/21834/28671089-af3664f6-72db-11e7-83c3-58def63e0e65.png)

What do you think? Also curious about your opinion on the slightly rounded edges, and if the explanatory diagram still conveys all the essentials of the original version – it's currently missing a few of the originals connector lines. And if you saw the version with cards and a variation of the bold color palette by @archiehicklin that I tried:

![image](https://cdn.discordapp.com/attachments/313693200269312003/339108267273879553/image.png)

> I think my favorite detail is the little emoji-like icons for each library. I need to steal that :)

Yeah, those little spots of an accent color work super well, although I'd like them to be a little more prominent somehow. They drown a little especially on ~72dpi-ish screens.

[I did steal them, literally, too](https://www.reactiflux.com/) – all props to @zachgibson who designed the [Reactiflux website](https://www.reactiflux.com/)! I mentioned that in the [PR description above](https://github.com/gatsbyjs/gatsby/pull/1605#issuecomment-317917387), but forgot to mention you @zachgibson explicitly. I hope it's okay to borrow!

DSchau commented:
@fk psh, nothing to be sorry about at all! I should've _not_ used that file to begin with, haha.

Should be remedied in the above PR!

zachgibson commented:
@fk For sure, borrow away! :)

KyleAMathews commented:
>It's pretty cool to see how my original 5-min mockup is turning into a proper, well-thought-out design under your hands! I wish every design project was this easy ;)

@SachaG @fk is a freelancer :-D

fk commented:
@Dschau 👍 👍 
@zachgibson 💪 Thank you ❤️  (your website too btw.)!
@KyleAMathews 🤗 😉

Responsive fixes coming…
